code/datums/components/callouts.dm

```diff
@@ -0,0 +1,153 @@
+#define CALLOUT_TIME (2.5 SECONDS)
+#define CALLOUT_COOLDOWN 3 SECONDS
+
+/// Component that allows its owner/owner's wearer to use callouts system - their pointing is replaced with a fancy radial which allows them to summon glowing markers
+/datum/component/callouts
+	/// If parent is clothing, slot on which this component activates
+	var/item_slot
+	/// Current user of this component
+	var/cur_user
+	/// Whenever the user should shout the voiceline
+	var/voiceline = FALSE
+	/// If voiceline is true, what prefix the user should use
+	var/radio_prefix = null
+	/// List of all callout options
+	var/static/list/callout_options = typecacheof(subtypesof(/datum/callout_option))
+	/// When last callout was used
+	var/last_callout_time = 0
+	/// Text displayed when parent is examined
+	var/examine_text = null
+
+/datum/component/callouts/Initialize(item_slot = null, voiceline = FALSE, radio_prefix = null, examine_text = null)
+	if (!isitem(parent) && !ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.item_slot = item_slot
+	src.voiceline = voiceline
+	src.radio_prefix = radio_prefix
+	src.examine_text = examine_text
+
+	if (ismob(parent))
+		cur_user = parent
+		return
+
+	var/atom/atom_parent = parent
+
+	if (!ismob(atom_parent.loc))
+		return
+
+	var/mob/user = atom_parent.loc
+	if (item_slot != null && user.get_item_by_slot(item_slot) != parent)
+		return
+
+	RegisterSignal(atom_parent.loc, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+	cur_user = atom_parent.loc
+
+/datum/component/callouts/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equipped))
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_dropped))
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examines))
+
+/datum/component/callouts/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_MOB_CLICKON, COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED, COMSIG_ATOM_EXAMINE))
+
+/datum/component/callouts/proc/on_equipped(datum/source, mob/equipper, slot)
+	SIGNAL_HANDLER
+
+	if (item_slot & slot)
+		RegisterSignal(equipper, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+		cur_user = equipper
+	else if (cur_user == equipper)
+		UnregisterSignal(cur_user, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+		cur_user = null
+
+/datum/component/callouts/proc/on_dropped(datum/source, mob/user)
+	SIGNAL_HANDLER
+
+	if (cur_user == user)
+		UnregisterSignal(cur_user, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+		cur_user = null
+
+/datum/component/callouts/proc/on_examines(mob/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	if (!isnull(examine_text))
+		examine_list += examine_text
+
+/datum/component/callouts/proc/on_click(mob/user, atom/clicked_atom, list/modifiers)
+	SIGNAL_HANDLER
+
+	if (!LAZYACCESS(modifiers, SHIFT_CLICK) || !LAZYACCESS(modifiers, MIDDLE_CLICK))
+		return
+
+	if (world.time < last_callout_time + CALLOUT_COOLDOWN)
```

Comment:
Also use the cooldown defines. `COOLDOWN_DECLARE`, `COOLDOWN_START`, `COOLDOWN_FINISHED`